docs/concepts/workloads/vm.md

```diff
@@ -1066,9 +1082,212 @@ spec:
         # The topology key is what designates the scope of the rule.
         # For example, "topologyKey: kubernetes.io/hostname" would
         # indicate the rule applies to nodes and not zones.
-        # For more detail, please refer to 
+        # For more detail, please refer to
         # https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/.
         topologyKey: topology.kubernetes.io/zone
 ```
 
+#### VM Anti-Affinity
+
+The following example shows VM anti-affinity at the host level, ensuring VMs with the label `tier: database` are distributed across different hosts:
+
+```yaml
+apiVersion: vmoperator.vmware.com/v1alpha5
+kind: VirtualMachine
+metadata:
+  name: database-vm-1
+  namespace: my-namespace-1
+  labels:
+    tier: database
+spec:
+  groupName: database-group  # Required for affinity rules
+  affinity:
+    vmAntiAffinity:
+      requiredDuringSchedulingPreferredDuringExecution:
+        labelSelector:
+        - matchLabels:
+            tier: database
+        topologyKey: kubernetes.io/hostname
+```
+
+## VirtualMachine Groups
+
+VirtualMachine Groups provide a way to manage multiple VMs as a single unit, enabling coordinated operations and advanced placement capabilities.
+
+### Group Membership
+
+A VM becomes a member of a group by setting its `spec.groupName` field to the name of a VirtualMachineGroup in the same namespace:
+
+```yaml
+apiVersion: vmoperator.vmware.com/v1alpha5
+kind: VirtualMachine
+metadata:
+  name: web-server-1
+  namespace: my-namespace
+spec:
+  groupName: web-tier-group  # Join the web-tier-group
+  className: my-vm-class
+  imageName: ubuntu-22.04
+  storageClass: my-storage-class
+```
+
+When a VM joins a group:
+- An owner reference to the group is automatically added to the VM
+- The VM appears in the group's `status.members` list
+- The VM inherits the group's power state management
+- The VM can use affinity rules (when `spec.groupName` is set)
+
+### VirtualMachineGroup Resource
+
+The VirtualMachineGroup resource defines how a collection of VMs should be managed:
+
+```yaml
+apiVersion: vmoperator.vmware.com/v1alpha4
+kind: VirtualMachineGroup
+metadata:
+  name: web-tier-group
+  namespace: my-namespace
+spec:
+  # Optional: This group can itself belong to another group
+  groupName: application-group
+
+  # Power state management for all group members
+  powerState: PoweredOn
+
+  # Boot order defines startup sequence
+  bootOrder:
+  - members:
+    - name: database-vm
+      kind: VirtualMachine
+    delay: 10s  # Wait 10 seconds before next group
+  - members:
+    - name: app-server-1
+      kind: VirtualMachine
+    - name: app-server-2
+      kind: VirtualMachine
+    delay: 5s
+  - members:
+    - name: web-server-1
+      kind: VirtualMachine
+    - name: web-server-2
+      kind: VirtualMachine
```

Comment:
Could be nice to change these to a VirtualMachineGroup kind member for the nested group kind member example.